### ISSUE TITLE
Training works great, but it doesn't computes correctly losses in validation data

### ISSUE THREAD

**Issue by AntonioVispi** [ultralytics/ultralytics#4805]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I trained a Yolov8 model for object detection with my custom dataset, made by videos. Observing at the results the network gave me (both the results.png and the results.csv), I noticed that the train losses are correct: they decrease with the training process, but the validation losses remain constant to zero from the beginning both for the val/box_loss and the val/dfl_loss, while the val/cls_loss is always infinite (it says “inf” in the results.csv).
![results](https://github.com/ultralytics/ultralytics/assets/102518682/b100bc49-b284-41e0-914f-38364d9250b9)

![resultsCSV](https://github.com/ultralytics/ultralytics/assets/102518682/1eec222a-b750-4654-99ab-c9e73be09a6a)
Just a few epochs, to show the issue about val/box_loss, val/dfl_loss, val/cls_loss.

This is the command line I used to train: 
`!python train.py model=yolov8n.yaml data='My_Path/data/data.yaml' epochs=50 imgsz=800 project='My_Path/Yolov8_models/Yolov8n'`
I also did some attempts like interchanging the train and validation datasets or using the train dataset both for training and for validation, but the error still persists, so I guess the problem it’s not on how I defined the validation dataset. Furthermore, I tried to use the trained network on an unseen video and it works just fine, so I think the problem is how it calculates the validation losses specifically (since the train losses are normally calculated), but I can’t find the error.

My suspect is that it computes the losses between ground truth and ground truth, so the loss is exactly zero.

Thank you in advance for your attention to my question, hope you will help me with my issue.

**Comment by glenn-jocher** [comment]
@AntonioVispi your dataset contains no validation labels, hence no validation metrics can be computed. You might want to review your dataset structure. See https://docs.ultralytics.com/datasets/detect/

**Comment by glenn-jocher** [comment]
@AntonioVispi 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. Validation loss plotting works correctly:

![image](https://github.com/ultralytics/ultralytics/assets/26833433/9c8a86fa-307c-4004-beff-7e3bb46bd610)



We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by AntonioVispi** [comment]
Thank you so much! My mistake. Now it works great!

**Comment by glenn-jocher** [comment]
You're welcome @AntonioVispi! I'm glad to hear that everything is working well now. Don't hesitate to ask if you have any more questions in the future. It's always important to us to ensure your experience with YOLOv8 is as smooth as possible. Happy training!
